core/admin/mailu/ui/views/users.py

```diff
@@ -93,12 +93,12 @@ def user_settings(user_email):
     form = forms.UserSettingsForm(obj=user)
     utils.formatCSVField(form.forward_destination)
     if form.validate_on_submit():
-        if form.forward_enabled.data and (form.forward_destination.data in ['', None] or type(form.forward_destination.data) is list):
+        user.forward_enabled = bool(flask.request.form.get('forward_enabled', False))
+        if form.forward_enabled and (form.forward_destination.data in ['', None] or type(form.forward_destination.data) is list):
             flask.flash('Destination email address is missing', 'error')
-            user.forward_enabled = True
-            return flask.render_template('user/settings.html', form=form, user=user)
-        if form.forward_enabled.data:
-            form.forward_destination.data = form.forward_destination.data.replace(" ","").split(",")
+            return flask.redirect(
+                flask.url_for('.user_settings', user_email=user_email))
+        form.forward_destination.data = form.forward_destination.data.replace(" ","").split(",")
```

Comment:
It's still ugly to convert to a comma separated string and then split into a list again while removing **all** whitespace and not only strip items, but that's like it was before, so I think it doesn't harm.

Reply:
I'm not sure I understand how we could do better here... at the time we didn't want to change the schema; maybe we can revisit this in master?

Reply:
Yes, that's what I wanted to say.